### PR TITLE
Migrate to pyproject.toml files

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**

### CODE REVIEW

File: pyproject.toml
Comment (GuillemBarroso) [inline pyproject.toml:8]: @PProfizi, how should we define ``version`` here? This is following other pyansys repos convention, but I remember we talked about PyDPF-Core using 0.7.Z.dev0, right?

File: pyproject.toml
Comment (GuillemBarroso) [inline pyproject.toml:43]:
```suggestion
Documentation = "https://dpf.docs.pyansys.com/"
```

File: pyproject.toml
Comment (GuillemBarroso) [inline pyproject.toml:42]:
```suggestion
Homepage = "https://dpf.docs.pyansys.com/"
```

**Review by GuillemBarroso — CHANGES_REQUESTED** [review]
Some minor changes. LGTM! Let's wait for @PProfizi review as well.

File: pyproject.toml
Comment (PProfizi) [reply pyproject.toml:8]: @GuillemBarroso @Revathyvenugopal162 Indeed, the version should be 0.7.3.dev0 (we are releasing 0.7.2)

Comment (PProfizi) [inline pyproject.toml:8]:
```suggestion
version = "0.7.3.dev0"
```

File: pyproject.toml
Comment (PProfizi) [inline pyproject.toml:11]:
```suggestion
requires-python = ">=3.7, <3.11"
```
Not sure this is required but it might be?

**Review by PProfizi — COMMENTED** [review]
I am alright with what I saw in the pyproject.toml
Do you know why black went and edited all files again? A change with the new version?
Also we might want to publish on testPyPI to make sure the metadata is alright.

File: pyproject.toml
Comment (Revathyvenugopal162) [reply pyproject.toml:8]: Ok i got your point. but usually we will not keep track of the patch releases, 0.7.dev0 or if we have already 0.7 release, then 0.8.dev0 is the usual way of keeping track, so that the patch release is not going to affect the dev version. https://peps.python.org/pep-0440/#developmental-releases. but no offence. ultimately the developers can decide .

**Comment by Revathyvenugopal162** [comment]
> I am alright with what I saw in the pyproject.toml Do you know why black went and edited all files again? A change with the new version? Also we might want to publish on testPyPI to make sure the metadata is alright.

The default line length formatting for black is 80. but in flake 8 we specified to 100, so i changed to 100. And i think it is a good number.

File: pyproject.toml
Comment (GuillemBarroso) [reply pyproject.toml]: I'm fine with either. I think it would be nice to be consistent with other repos and use 0.8.dev0, but I am not fully aware of the implications of changing this. @PProfizi, your call.

**Comment by GuillemBarroso** [comment]
> > I am alright with what I saw in the pyproject.toml Do you know why black went and edited all files again? A change with the new version? Also we might want to publish on testPyPI to make sure the metadata is alright.
> 
> The default line length formatting for black is 80. but in flake 8 we specified to 100, so i changed to 100. And i think it is a good number.

Yep. I like 100.

**Review by PProfizi — APPROVED** [review]

**Review by PProfizi — APPROVED** [review]

**Pull request merged**
